File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.java
class Solution {
    public int solution(String word) {
        String words="AEIOU";
        int[] x={781,156,31,6,1};
        int index,result= word.length();
        for(int i=0;i<word.length();i++){
          index=words.indexOf(word.charAt(i));
          result+=x[i]*index;
        }
        System.out.println(result);
        return result;
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
# def solution(word):
#     answer = 0
#     alpha  = ["A","E","I","O","U",""]
#     ans = []
#     for i in alpha:
#         for j in alpha:
#             for k in alpha:
#                 for l in alpha:
#                     for m in alpha:
#                         w = i+j+k+l+m
#                         if w not in ans: ans.append(w)
#     ans.sort()
#     answer = ans.index(word)
#     return answer

def solution(word):
    answer = 0
    for i, n in enumerate(word):
        answer += (5 ** (5 - i) - 1) / (5 - 1) * "AEIOU".index(n) + 1
    return answer

